iceoryx_examples/ara_com/ara/types.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_ARA_COM_TYPES_HPP
+#define IOX_EXAMPLES_ARA_COM_TYPES_HPP
+
+#include "iceoryx_hoofs/cxx/function.hpp"
+#include "iceoryx_hoofs/cxx/optional.hpp"
+#include "iceoryx_hoofs/cxx/string.hpp"
+#include "iceoryx_posh/capro/service_description.hpp"
+#include "iceoryx_posh/popo/sample.hpp"
+
+#include "com/sample_allocatee_ptr.hpp"
+
+#include <future>
+#include <memory>
+
+namespace ara
+{
+class Runtime;
+namespace core
+{
+template <typename T>
+using Result = T;
+
+using String = iox::cxx::string<100>;
```

Comment:
The runtime is using a `std::string`. We should only use one string which we define here I think. Probably std::string, as it is easier to convert into `cxx::string` but not ideal (wrt. efficiency and code elegance).

Reply:
For a rough example that's perfectly fine for me. For a proper version one needs to provide a conversion function between `std::string` and `cxx::string` or extend `cxx::string` to their needs.